Fix intermittent playback through browser when force headphone output.

Ken-Vamrs opened the pull request.

Ken-Vamrs committed:
codec: es8316: update dapm widgets and routes

Fix intermittent playback through browser when force headphone output.

Signed-off-by: Ken Wang <ken@radxa.com>

RadxaYuntian commented on sound/soc/codecs/es8316.c:
```diff
@@ -232,157 +232,143 @@ static const struct snd_kcontrol_new es8316_dacsrc_mux_controls =
-
-	SND_SOC_DAPM_OUTPUT("HPOL"),
-	SND_SOC_DAPM_OUTPUT("HPOR"),
+    /* Input Lines */
```
缩进不对，看不出你到底改了啥

Ken-Vamrs replied:
从5.10内核抄过来的，差别有点大，对着改没改好。
优化了不断暂停播放问题，以及上电时的哒哒声

RadxaYuntian replied:
其实可能是我们的alsa config的问题：https://github.com/radxa-pkg/radxa-alsa-config/blob/main/src/alsa/ucm2/conf.d/rockchip-es8316/HiFi.conf

RadxaYuntian replied:
可能需要参考一下这：https://github.com/alsa-project/alsa-ucm-conf/tree/master/ucm2/Rockchip/es8316
修改这里用上游的config看看：https://github.com/radxa/kernel/pull/372/files#diff-c3cd5181d410bdbe1149937d61d39c8a36b22851f50caf8e22d54ba114245196R118

Ken-Vamrs replied:
天哥 这个修改没有效果，不得行

RadxaYuntian replied:
找armbian的看看？他们也有支持我们用es8316的设备吧

Ken-Vamrs replied:
armbian 支持，也没有复现到，但是他不是用rockchip ucm2控制的

RadxaYuntian replied:
我们也换成他们那种配置行么

Ken-Vamrs replied:
这个能不能合入呢，测试时间长，没有复现

RadxaYuntian replied:
我就感觉我们以后就一直要用这个旧的代码，而不是看问题到底在哪

RadxaYuntian replied:
@RadxaStephen 看看到底是不是就先合入这个旧代码了

RadxaYuntian commented on arch/arm64/boot/dts/rockchip/rk3588-rock-5t.dts:
```diff
@@ -220,7 +220,7 @@
 
 		user-led2 {
-			gpios = <&gpio1 RK_PA7 GPIO_ACTIVE_HIGH>;
+			gpios = <&gpio0 RK_PA0 GPIO_ACTIVE_HIGH>;
```
这是硬件改版还是写错了？

Ken-Vamrs replied:
硬件改版

RadxaYuntian replied:
之前的版本有出售过么？

Ken-Vamrs replied:
打样阶段  应该只是送过样品

RadxaYuntian commented on arch/arm64/boot/dts/rockchip/rk3588-rock-5t.dts:
```diff
@@ -1196,7 +1196,7 @@
 		"", "", "PIN_31", "PIN_29",
 		/* GPIO1_A4-A7 */
-		"", "", "", "",
+		"", "", "", "PIN_37",
```
这是硬件改版还是之前忘了？

RadxaStephen commented:
Commits for LED, 40-PIn and rtc are merged.

Ken-Vamrs committed:
arm64: dts: rockchip: modify fan pwm period to 60us

Reduced pwm frequency to 16.6k Hz for a larger adjustable range of AO3416 mosfet.

Signed-off-by: Ken Wang <ken@radxa.com>

RadxaStephen commented:
Merged.
